app/models/interactive_page.rb

```diff
@@ -118,24 +118,37 @@ def interactive_page_items
     page_items.select{ |pi| Embeddable::is_interactive?(pi.embeddable) }
   end
 
-  def section_embeddables(section_title)
-    section = sections.find { |s| s.title == section_title }
-    if section
-      section.page_items.map { |i| i.embeddable }
-    else
-      []
+  def section_embeddables(section)
+    # For now, we continue to support passing a section name as a string to 
+    # this method, but this legacy feature will be removed in the future and 
+    # passing the section itself will become the only option.
+    if !section.instance_of? Section
+      section = sections.find { |s| s.title == section }
     end
-  end
 
-  def main_embeddables
-    # Embeddables that do not have section specified (nil section).
-    section_embeddables(Section::DEFAULT_SECTION_TITLE)
+    section_embeddables = []
+    if section
+      # If a section's primary column is on the right, we need to adjust the 
+      # embeddables' order so they're output in the same order they appear on 
+      # the activity page.
+      primary_right_layouts = ["40-60", "30-70", "responsive-2-columns"]
+      if primary_right_layouts.include? section.layout
+        secondary_column_items = section.page_items.where(:column => "secondary").order(:position)
+        primary_column_items = section.page_items.where(:column => "primary").order(:position)
+        page_items = secondary_column_items + primary_column_items
+        section_embeddables = page_items.map { |i| i.embeddable }
+      else
+        section_embeddables = section.page_items.map { |i| i.embeddable }
+      end
+    end
+    section_embeddables
   end
 
   def visible_embeddables
     results = []
     sections.each do |s|
-      results += section_visible_embeddables(s.title)
+      puts "LKLKLKLK 8 #{s.inspect}"
```

Comment:
probably should remove this debugging statement

Reply:
Whoops. Yeah, probably :)